### DISCUSSION TITLE
Duplicate Extra Lives on Score Tally?

### DISCUSSION THREAD

**Discussion by CCShakeCat** [Techokami/SonicWorldsNext#217] [Q&A]
So I was trying to make the score tallying more accurate to the MD/Genesis games and I have no idea what is happening that is giving Sonic lots of extra lives. Is there a way to fix this?

[Here is some footage I have that will show you what I am talking about](https://youtu.be/KmKuELGQq10)

Update 12/22 2024: It seems to have fixed itself, but I will keep this open if it happens again

**Comment by r6915ee** [comment]
In such cases you'll likely need to show your actual code in case you changed it, because there's not really much behind the scenes that can be seen here.

**Reply by CCShakeCat** [reply]
![Screenshot_20241223_082607_Firefox](https://github.com/user-attachments/assets/e2239782-87cb-408b-a13f-6b8cff5ef426)
Basically i made the Time Bonus decrease by 1,000 if over 10,000, then decrease by 100 if over 1,000

Like I said in the edit, a problem seem to have fixed itself, but until something arises, this will remain open

**Reply by r6915ee** [reply]
Something might be repeating the addition of lives then, though I'm not sure exactly what is causing it.

**Reply by DimensionWarped** [reply]
That screenshot is borderline unreadable.

EDIT: I'd suggest only doing one of those conditions in a given frame.

Also, following the player refactor (which is still probably at least another week or two out) PR, there will be a player function to latch onto here for adding score rather than going through it in this fashion.
